pgjdbc/src/test/java/org/postgresql/test/jdbc42/TimestampUtilsTest.java

```diff
@@ -62,6 +63,46 @@ public void testToLocalTime() throws SQLException {
     assertEquals(LocalTime.parse("23:59:59.999999999"), timestampUtils.toLocalTime("24:00:00"));
   }
 
+  @Test
+  public void testToStringOfOffsetTime() {
+    TimestampUtils timestampUtils = createTimestampUtils();
+
+    assertEquals("00:00:00+00", timestampUtils.toString(OffsetTime.parse("00:00:00+00:00")));
```

Comment:
Would you please factor it into a `assertOffsetTimeToString(String offsetTime)` method?

Then the test code would be easier to read (less `timestampUtils.toString(OffsetTime.parse` repetitions), and the assertion messages would be easier to understand: the failure message could include the reason of the failure so CI failures are easier to understand.

For instance:


```java
void assertOffsetTimeToString(String inputTime, String expectedOutput) {
  // Note how failure message would include **input** value, so it would be easier to understand what went wrong.
  assertEquals(
    expectedOutput,
    timestampUtils.toString(OffsetTime.parse(inputTime)),
    () -> "timestampUtils.toString(OffsetTime.parse(" + inputTime + "))"
  );
}
```

An alternative option is to use `@ParameterizedTest`, however, most of our tests are still in JUnit4, so adding a method seems to be the way to go.

Reply:
This is again copypaste unchanged from the corresponding LocalTime test. I did not want to change the pattern of other tests in that class.
Your ideas are good, but that should also be applied to the other tests in that class. But that's out of scope of this issue. Having all of them working the same way makes them easier to change later to use parameters.

Reply:
Do we agree to not change this here and better make a cleanup of the whole `TimestampUtilsTest` class? I would like to resolve this here and won't fix.

Reply:
I would not use parameters here, lets just add a method for the verification as suggested. Will do this also for LocalTime!

Reply:
Hi, I rewrote the test with the pattern provided above (a separate assertion method). Because of this the comments about nanos are not needed because message says what is expected

Reply:
The comments regarding "499 vs 500" would still be helpful for the edge cases.
It might make sense to have an overload like `assertToStringOfLocalTimeUnchanged(String expectedOutput) { assertToStringOfLocalTime(expectedOutput, expectedOutput);`

Reply:
yes, i will readd the comments. I was thinking about adding them to the error message, but that seemed too much. I just have to revert.

Adding an overload is a good idea, not sure if this is really needed here. But it would bring the attention of reader directly to the "special" cases.

Reply:
I won't add the overloads. Although you have duplicated strings the test looks more clear to me in its current state.

Comments were added back.

Reply:
The other reason why I don't want to add this "unchanged": The formats by postgres and Java have a different definition of the formatting. Postgres uses a shorter zoneoffset notation. We actually want to test here how a LocalTime is converted to the postgres representation. That some of the results are identical to LocalTime.toString() is just happening but no requirement. For OffsetTime that's more obvious because of different timezone notation.